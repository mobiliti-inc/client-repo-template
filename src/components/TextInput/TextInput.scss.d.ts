// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'checkmark-label': string;
  'dark': string;
  'error': string;
  'error-message': string;
  'fades-in-out': string;
  'input-base': string;
  'input-date': string;
  'input-email': string;
  'input-file': string;
  'input-group': string;
  'input-image': string;
  'input-label': string;
  'input-number': string;
  'input-password': string;
  'input-tel': string;
  'input-text': string;
  'input-url': string;
  'password-visibility-indicator': string;
}
declare var cssExports: CssExports;
export = cssExports;
