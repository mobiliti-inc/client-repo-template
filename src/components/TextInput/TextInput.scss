@import "../../styles/_variables.scss";

.input-group {
  position: relative;
  display: flex;
  flex-direction: column;
}

.input-base {
  height: 50px;
  width: 100%;
  font-size: 22px;
  font-family: inherit;
  border: none;
  transition: all .5s;
  border-bottom: 1px solid rgba(29, 49, 85, 0.2);
  text-transform: capitalize;
  font-family: $hanken-sans;

  &:focus {
    outline: none;
    border-bottom: 3px solid $blue;
  }

  &::-webkit-input-placeholder {
    color: $darker-blue;
    opacity: 0.6;
  }

  &:placeholder-shown+.input-label {
    opacity: 0;
    visibility: hidden;
    transform: translateY(-40px);
  }

  &.error {
    border-bottom: 3px solid $fast-car-red;
  }
}

.input-email,
.input-password,
.input-number,
.input-image,
.input-text,
.input-file,
.input-date,
.input-url,
.input-tel {
  composes: input-base;
  text-transform: none;
}

.input-label {
  display: block;
  font-size: 12px;
  transform: translateY(-60px);
  color: $darker-blue;
  opacity: 0.6;
  font-family: $hanken-sans;
  line-height: 14px;
  text-transform: capitalize;
  transition: all .5s;
}

.checkmark-label {
  display: inline-block;
  position: absolute;
  top: 0px;
  right: 0px;
  height: 20px;
  width: 12px;
  border: solid $green-check;
  border-width: 0 2px 2px 0;
  cursor: pointer;
  transform: rotate(45deg);
}

.dark {
  color: $black;
  border-color: $black;

  &::placeholder {
    color: $black;
  }

  label {
    color: $black;
  }
}

.error-message {
  color: $fast-car-red;
  height: 14px;
  font-family: $hanken-sans;
  margin-top: 5px;
  font-size: 12px;
  line-height: 14px;
}

.password-visibility-indicator {
  display: inline-block;
  position: absolute;
  top: 15px;
  right: 0;
  line-height: 19px;
  color: $blue;
  font-family: $hanken-sans;
  font-size: 16px;
  border: none;
  background-color: transparent;
  text-transform: capitalize;
  cursor: pointer;

  &:active,
  &:focus {
    outline: none;
  }
}