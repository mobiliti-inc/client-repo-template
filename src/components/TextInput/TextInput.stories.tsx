import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import * as React from 'react';
import { TextInput } from '../';

storiesOf('TextInput', module)
	.add('default', () => (<div style={{ width: "300px" }}><TextInput type={text('type', 'text')}
		onChange={() => { }}
		id="text"
		value={text('value', '')}
		placeholder={text('value', '')}
		disabled={boolean('disabled', false)}
		isValid={boolean('isValid', false)}
		dark={boolean('dark', false)} /></div>
	)
	);