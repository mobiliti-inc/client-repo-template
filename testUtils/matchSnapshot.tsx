const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
