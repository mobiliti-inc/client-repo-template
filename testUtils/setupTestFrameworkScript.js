const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
