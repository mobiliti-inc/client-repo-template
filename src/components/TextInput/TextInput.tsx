import * as React from 'react';
import cx from 'classnames';

import './TextInput.scss';

export const INPUT_TYPES = {
	PASSWORD: 'password',
	NUMBER: 'number',
	EMAIL: 'email',
	IMAGE: 'image',
	TEXT: 'text',
	FILE: 'file',
	DATE: 'date',
	PHONE: 'tel',
	URL: 'url'
};

interface TextInputProps {
	type: string;
	onChange: (...args: any[]) => any;
	value: string;
	id: string;
	placeholder?: string;
	disabled?: boolean;
	isValid: boolean;
	inputDidChange?: boolean;
	onBlur?: (...args: any[]) => any;
	label?: string;
	name?: string;
	dark?: boolean;
	className?: string;
	errorMessage?: React.ReactNode | string;
	shouldShowVisibility?: boolean;
	changePasswordVisibilityStatus?: (...args: any[]) => any;
}

const TextInput: React.FC<TextInputProps> = (props) => {
	const {
		placeholder,
		type,
		onChange,
		disabled,
		label,
		value,
		name,
		id,
		isValid,
		dark,
		className,
		inputDidChange,
		onBlur,
		errorMessage
	} = props;

	const [inputValue, setInputValue] = React.useState('');

	React.useEffect(() => {
		setInputValue(value);
	}, [value]);

	const baseStyles = cx(`input-${type}`, {
		error: !isValid
	});

	const styleName = cx(baseStyles, { dark }, className);

	// TODO: We really shouldn't use any here ðŸ˜“ but rather React.FormEvent<HTMLInputElement>, value was kinda nagging us
	const handleChange = (event: any) => {
		setInputValue(event.target.value);
		onChange(event);
	};

	return (
		<React.Fragment>
			<div styleName="input-group">
				<input
					placeholder={placeholder}
					styleName={styleName}
					className={className}
					disabled={disabled}
					onChange={(event) => handleChange(event)}
					type={type}
					value={inputValue}
					name={name}
					id={id}
					onBlur={onBlur}
				/>
				{label && (
					<label styleName="input-label" htmlFor={id}>
						{label}
					</label>
				)}
				{isValid && inputDidChange && value && value.length > 0 && <label styleName="checkmark-label" />}
				{!isValid && errorMessage && <span styleName="error-message">{errorMessage}</span>}
			</div>
		</React.Fragment>
	);
};

export default TextInput;
