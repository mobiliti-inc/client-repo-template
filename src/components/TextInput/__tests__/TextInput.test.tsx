import React from 'react';

import { matchSnapshot } from '../../../../testUtils/matchSnapshot';
import { shallow } from 'enzyme';

import TextInput, { INPUT_TYPES } from '../TextInput';

describe('TextInput', () => {
	it('renders', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				isValid
				value="test"
				id="test-input"
			/>
		);

		matchSnapshot(tree);
	});

	it('renders the text input with an error class when not valid', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				isValid={false}
				value="test"
				id="test-input"
			/>
		);
		expect(tree.find('input.error').length).toBe(1);
	});

	// it('sets the correct value', () => {
	// 	const subject = shallow(
	// 		<TextInput
	// 			type={INPUT_TYPES.TEXT}
	// 			onChange={() => { }}
	// 			value=""
	// 			isValid
	// 			id="test-input"
	// 		/>
	// 	);

	// 	subject.setProps({ value: 'test' });
	// 	expect(subject.instance().props.value).toEqual('test');
	// });

	it('renders the text input with a label when focused by setting a value', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				value=""
				isValid
				id="test-input"
				label="Text"
			/>
		);

		tree.setProps = { value: 'test' };
		matchSnapshot(tree);
	});

	it('renders the text input with a checkmark at end when input field is valid and inputDidChange is true', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				value="hello"
				id="test-input"
				label="Text"
				isValid
				inputDidChange
			/>
		);

		tree.setProps = { value: 'test' };
		expect(tree.find('.checkmark-label').length).toBe(1);
		matchSnapshot(tree);
	});

	it('does not render the error message when its not provided', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				value=""
				id="test-input"
				label="Text"
				isValid={false}
				inputDidChange
			/>
		);

		expect(tree.find('.error-message').length).toBe(0);
	});

	it('renders the error message prop when not valid and inputDidChange is false', () => {
		const tree = shallow(
			<TextInput
				type={INPUT_TYPES.TEXT}
				onChange={() => { }}
				value=""
				id="test-input"
				label="Text"
				isValid={false}
				inputDidChange={false}
				errorMessage="Error"
			/>
		);

		expect(tree.find('.error-message').length).toBe(1);
	});

	describe('Email Text Input', () => {
		it('renders with an error class only when invalid', () => {
			const tree = shallow(
				<TextInput
					type={INPUT_TYPES.EMAIL}
					onChange={jest.fn}
					isValid={false}
					value="test"
					id="test-input"
					errorMessage="Invalid"
				/>
			);
			expect(tree.find('input.error').length).toBe(1);
		});

		it('renders without an error class when valid', () => {
			const tree = shallow(
				<TextInput
					type={INPUT_TYPES.EMAIL}
					onChange={jest.fn}
					isValid
					value="test"
					id="test-input"
					errorMessage=""
				/>
			);
			expect(tree.find('input.error').length).toBe(0);
		});
	});

	describe('Password Text Input', () => {
		// it('renders with password-visibility-indicator class only when shouldShowVisibility is true and value length is more than one', () => {
		// 	const tree = shallow(
		// 		<TextInput
		// 			type={INPUT_TYPES.PASSWORD}
		// 			onChange={jest.fn}
		// 			isValid={false}
		// 			value="test"
		// 			id="test-input"
		// 			shouldShowVisibility
		// 		/>
		// 	);
		// 	expect(tree.find('.password-visibility-indicator').length).toBe(1);
		// });

		it('renders without password-visibility-label class when shouldShowVisibility is false', () => {
			const tree = shallow(
				<TextInput
					type={INPUT_TYPES.PASSWORD}
					onChange={jest.fn}
					isValid={false}
					value="test"
					id="test-input"
				/>
			);
			expect(tree.find('.password-visibility-indicator').length).toBe(0);
		});
	});
});
