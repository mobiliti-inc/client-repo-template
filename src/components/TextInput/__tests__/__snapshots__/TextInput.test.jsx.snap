// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders 1`] = `
<React.Fragment>
  <div
    className="input-group"
  >
    <input
      className="input-text"
      disabled={false}
      id="test-input"
      name=""
      onBlur={null}
      onChange={[Function]}
      placeholder=""
      type="text"
      value="test"
    />
  </div>
</React.Fragment>
`;

exports[`TextInput renders the text input with a checkmark at end when input field is valid and inputDidChange is true 1`] = `
<React.Fragment>
  <div
    className="input-group"
  >
    <input
      className="input-text"
      disabled={false}
      id="test-input"
      name=""
      onBlur={null}
      onChange={[Function]}
      placeholder=""
      type="text"
      value="hello"
    />
    <label
      className="input-label"
      htmlFor="test-input"
    >
      Text
    </label>
    <label
      className="checkmark-label"
    />
  </div>
</React.Fragment>
`;

exports[`TextInput renders the text input with a label when focused by setting a value 1`] = `
<React.Fragment>
  <div
    className="input-group"
  >
    <input
      className="input-text"
      disabled={false}
      id="test-input"
      name=""
      onBlur={null}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <label
      className="input-label"
      htmlFor="test-input"
    >
      Text
    </label>
  </div>
</React.Fragment>
`;
